integration-templates/algolia/algolia-create-contacts.ts

```diff
@@ -0,0 +1,33 @@
+import type { NangoAction, AlgoliaContact, AlgoliaCreateContactInput } from './models';
+
+export default async function runAction(nango: NangoAction, input: AlgoliaCreateContactInput): Promise<AlgoliaContact> {
+    if (!input.name) {
+        throw new nango.ActionError({
+            message: 'name is a required field'
+        });
+    }
+
+    const endpoint = `/1/indexes/contacts`;
+
+    try {
+        const postData = {
+            name: input.name,
+            company: input.company,
+            email: input.email
+        };
+        const resp = await nango.post({
```

Comment:
Note: the Algolia API in itself is not fully resilient; the sdks contains a little bit of logic to target different servers in case of downtime. We probably don't care but for power user they will need to load the `algoliasearch` npm package somehow